runtime/src/iree/hal/drivers/hip/pending_queue_actions.c

```diff
@@ -750,9 +751,14 @@ iree_status_t iree_hal_hip_pending_queue_actions_issue(
           continue;
         }
 
-        // Try to acquire a HIP event from an existing device wait timepoint.
+        // Try to acquire a HIP event from an existing device signal timepoint.
         // If so, we can use that event to wait on the device.
         // Otherwise, this action is still not ready for execution.
+        // A HIP event can not be waited on before it is scheduled for
+        // recording.
+        // If not schedule yet, it is as if already occurred.
```

Comment:
.. before it is recorded into the steam--before recording, event represents an empty set of work so waiting on it will just return success. So here we must guarantee the HIP event is indeed recorded, which means it's associated with some device signal timepoint on the semaphore timeline.

Reply:
I changed the comment. It is not exactly as you suggested.